tests/foreman/ui/test_errata.py

```diff
@@ -191,176 +136,165 @@ def cv_publish_promote(sat, org, cv, lce=None, needs_publish=True):
-    assert len(repos) > 0, 'Passed request must be a non-empty list.'
-
-    custom_products = []
+    params = getattr(request, 'param', None)
+    environment = module_lce
+    if params is None:
+        repos = []
+    else:
+        if any(p.lower() == 'library' for p in params):
+            environment = 'Library'
+        repos = [p for p in params if str(p).lower() != 'library']
+
+    if rhel_contenthost.subscribed:
+        rhel_contenthost.unregister()
     custom_repos = []
     for repo_url in repos:
-        # Publishes a new cvv, associates org, ak, cv, with custom repo:
-        custom_repo_info = module_target_sat.cli_factory.setup_org_for_a_custom_repo(
-            {
-                'url': repo_url,
-                'organization-id': module_sca_manifest_org.id,
-                'lifecycle-environment-id': module_lce.id,
-                'activationkey-id': errata_host_ak.id,
-                'content-view-id': module_cv.id,
-            }
-        )
-        custom_products.append(custom_repo_info['product-id'])
-        custom_repos.append(custom_repo_info['repository-id'])
-
-    # Publish new version and promote with all content
-    cv_publish_promote(module_target_sat, module_sca_manifest_org, module_cv, module_lce)
-    result = rhel_contenthost.register(
-        activation_keys=errata_host_ak.name,
-        target=module_target_sat,
-        org=module_sca_manifest_org,
-        loc=None,
-    )
-    assert result.status == 0, f'Failed to register host:\n{result.stderr}'
-    assert rhel_contenthost.subscribed, (
-        f'Failed to subscribe host to content, host: {rhel_contenthost.hostname}'
-        f' Attempting to subscribe to content-view id: {module_cv.id}'
-        f' Using activation-key id: {errata_host_ak.id}'
+        new_repo = module_target_sat.api.Repository(url=repo_url, product=module_product).create()
+        new_repo.sync()
+        custom_repos.append(new_repo)
+    if len(custom_repos) > 0:
+        module_cv.repository = custom_repos
+        module_cv.update(['repository'])
+    # Publish/promote CV if needed, associate entities, register client:
+    setup = module_target_sat.api_factory.register_host_and_needed_setup(
+        organization=module_sca_manifest_org,
+        client=rhel_contenthost,
+        activation_key=module_ak,
+        environment=environment,
+        content_view=module_cv,
     )
+    assert setup['result'] != 'error', f'{setup["message"]}'
+    assert (client := setup['client'])
+    assert client.subscribed
+    # nothing applicable to start
+    result = client.execute('subscription-manager repos')
+    assert client.applicable_errata_count == 0
+    assert client.applicable_package_count == 0
+    # no repos given/present, subscription-manager should report error
+    if len(repos) == 0:
+        assert client.execute(r'subscription-manager repos --enable \*').status == 1
```

Comment:
hmm but if sub-man fails to find repos to enable, then that will probably mean errata applicability won't work either, correct?

Reply:
Yes, but here we just assert that it really fails when no repos provided.
If `len(repos) == 0` means that we have no repos where we would install or apply errata from anyway. So is it allowed variant?

Reply:
gotcha yeah I see, code is a little different but my thought process was this;
if we want to register host with no repos at time of reg 
(allowed & default variant of the chost fixture)...
I wanted to ensure sub-manager also found no repos to enable. 
Seems unnecessary though?

Reply:
Because after registration, we can always add repos to cv and publish/promote, and use subman as normal after. So by default I also wanted to check subman found nothing

Reply:
Ok, gotcha. It just appeared odd to me that we check sub-man functionality here, not Sat (like do we want to fail Satellite test when sub-man is broken actually). But we rely on `sub-man repos` and others anyway, so I don't mind to keep it.

Reply:
@vsedmik Should we also check some satellite functionality too? Such as making sure no repos enabled in AK, when none are specified?

Reply:
That would rather be a negative test case for the Activation-key component. I wouldn't put it in Errata testing.